src/v/cluster/topics_frontend.cc

```diff
@@ -527,46 +533,53 @@ static allocation_request make_allocation_request(
     return req;
 }
 
-errc topics_frontend::validate_topic_configuration(
+topic_result topics_frontend::validate_topic_configuration(
   const custom_assignable_topic_configuration& assignable_config) {
+    const auto make_result = [&assignable_config](
+                               errc err,
+                               std::optional<ss::sstring> msg = std::nullopt) {
+        return cluster::make_error_result(
+          assignable_config.cfg.tp_ns, err, std::move(msg));
+    };
     if (!validate_topic_name(assignable_config.cfg.tp_ns)) {
-        return errc::invalid_topic_name;
+        return make_result(errc::invalid_topic_name);
```

Comment:
Would it be worth adding error messages here as well? That way we can remove some logic from the rpk and keep it centralised here.

Reply:
> Would it be worth adding error messages here as well? That way we can remove some logic from the rpk and keep it centralised here.

There are probably many places that could be improved, I'd prefer to keep this PR focused on the sanctions, though.

Reply:
Sorry, I should have been more explicit. This PR is good to go :+1: 
It was more of a general question.

Reply:
You get your wish